test/scaling.jl

```diff
@@ -0,0 +1,435 @@
+                @test Transforms.apply(M, scaling; dims=1, inds=[2]) ≈ expected atol=1e-5
+
+                scaling = MeanStdScaling(M; dims=2)
+                expected = [-1.0 0.0]'
+                @test Transforms.apply(M, scaling; dims=2, inds=[2]) ≈ expected atol=1e-5
+            end
+
+            @testset "Re-apply" begin
+                scaling = MeanStdScaling(A; dims=2)
+                Transforms.apply(A, scaling; dims=2)
+
+                # Expect scaling parameters to be fixed to the first data applied to
+                A2 = [1.0 -2.0 -1.0; 0.5 0.0 0.5]
+                A2_expected = [2.0 -4.0 -2.0; -0.5 -1.0 -0.5]
+
+                @test Transforms.apply(A2, scaling; dims=2) ≈ A2_expected atol=1e-5
+            end
+
+            @testset "Inverse" begin
+                scaling = MeanStdScaling(A)
+                transformed = Transforms.apply(A, scaling)
+                inverted = Transforms.apply(transformed, scaling; inverse=true)
+
+                @test transformed ≈ A_expected atol=1e-5
+                @test inverted ≈ A atol=1e-5
+            end
+        end
+
+        @testset "AxisKey" begin
+            M = [0.0 -0.5 0.5; 0.0 1.0 2.0]
+            A = KeyedArray(M; foo=["a", "b"], bar=["x", "y", "z"])
+            M_expected = [-0.559017 -1.11803 0.0; -0.559017 0.559017 1.67705]
+            A_expected = KeyedArray(M_expected; foo=["a", "b"], bar=["x", "y", "z"])
+
+            @testset "Non-mutating" begin
+                scaling = MeanStdScaling(A)
+                @test Transforms.apply(A, scaling) ≈ A_expected atol=1e-5
+                @test scaling(A) ≈ A_expected atol=1e-5
+
+                # Test the transform was not mutating
+                @test !isapprox(A, A_expected; atol=1e-5)
+            end
+
+            @testset "Mutating" begin
+                scaling = MeanStdScaling(A)
+                _A = copy(A)
+                Transforms.apply!(_A, scaling)
+                @test _A isa KeyedArray
+                @test _A ≈ A_expected atol=1e-5
+            end
+
+            @testset "dims = :" begin
+                scaling = MeanStdScaling(A)
+                @test Transforms.apply(A, scaling; dims=:) ≈ A_expected atol=1e-5
+            end
+
+            @testset "dims = 1" begin
+                scaling = MeanStdScaling(A; dims=1)
+                A_2_expected = [0.0 -0.707107 -0.707107; 0.0 0.707107 0.707107]
+                @test Transforms.apply(A, scaling; dims=1) ≈ A_2_expected atol=1e-5
+            end
+
+            @testset "dims = 2" begin
+                scaling = MeanStdScaling(A; dims=2)
+                A_1_expected = [0.0 -1.0 1.0; -1.0 0.0 1.0]
+                @test Transforms.apply(A, scaling; dims=2) ≈ A_1_expected atol=1e-5
+            end
+
+            @testset "inds" begin
+                scaling = MeanStdScaling(M)
+                expected = [-0.559017, -1.11803]
+                @test Transforms.apply(M, scaling; inds=[2, 3]) ≈ expected atol=1e-5
```

Comment:
Just for readability, everywhere `expected` is a one-off can we subsitute it for the actual values

```julia
@test Transforms.apply(M, scaling; inds=[2, 3]) ≈ [-0.559017, -1.11803] atol=1e-5
```

Reply:
I think I did this because of line limits, so should I break the statement into multiple lines?

Reply:
The BlueStyle/Invenia Julia line limits are recommendations more than hard rules. I think in this case for readability is one where having it go over the line length on one line is more valuable than adhering to line length